src/Money.php

```diff
@@ -49,20 +49,24 @@
     ];
 
     /**
-     * @param int|string $amount   Amount, expressed in the smallest units of $currency (eg cents)
-     * @param Currency   $currency
+     * @param float|int|string $amount   Amount, expressed in the smallest units of $currency (eg cents)
+     * @param Currency         $currency
      *
      * @throws \InvalidArgumentException If amount is not integer
      */
     public function __construct($amount, Currency $currency)
     {
         if (filter_var($amount, FILTER_VALIDATE_INT) === false) {
-            $numberFromString = Number::fromString($amount);
-            if (!$numberFromString->isInteger()) {
+            if (is_float($amount)) {
```

Comment:
Not sure this will work as intended. What if `$amount == (float)PHP_INT_MAX+1` ?

Reply:
this works fine.

Reply:
sorry, i misread. yes this doesn't work. what should be the behaviour/error here?